policy/aggregation_id_compress.go

```diff
@@ -79,27 +81,33 @@ func NewAggregationTypeDecompressor() AggregationIDDecompressor {
 	return &aggregationIDDecompressor{
 		bs:  bs,
 		buf: bs.Bytes(),
-		res: make([]AggregationType, 0, totalAggregationTypes),
 	}
 }
 
-func (c *aggregationIDDecompressor) Decompress(id AggregationID) (AggregationTypes, error) {
+func (c *aggregationIDDecompressor) Decompress(pool AggregationTypesPool, id AggregationID) (AggregationTypes, error) {
+	// NB(cw) it's guaranteed that len(c.buf) == len(id) == AggregationIDLen, we need to copy
+	// the words from id into a slice to be used in bitset
 	for i := range id {
 		c.buf[i] = id[i]
 	}
 
 	c.bs.Reset(c.buf)
```

Comment:
Actually, this is interesting, the main reason I added the Reset function to bitset is so it can reset all the internal data correctly, including the underlying slice and a length, so if you want to reset the bitset to use a slice with different size, the bitset.length is updated accordingly. 

But for our case the length of the slice will always be the AggregationIDLen, we can get away with copying the ints to the c.buf without changing the bitset.length. Gonna remove my bitset branch from it.